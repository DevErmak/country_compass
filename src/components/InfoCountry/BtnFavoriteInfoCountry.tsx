import React from 'react';
import { getIsAuthentication, getListFavoriteCountries } from '../../store/user/userSelectors';
import { useSelector } from 'react-redux';
import {
  addFavoriteCountry,
  removeFavoriteCountry,
  setModal,
} from '../../store/user/infoUserSlice';
import { useDispatch } from 'react-redux';

import { BsFillStarFill } from 'react-icons/bs';

import { BsStar } from 'react-icons/bs';

type Props = { nameCountry: string };

export default function BtnFavoriteInfoCountry({ nameCountry }: Props) {
  const dispatch = useDispatch();
  const listFavoriteCountries = useSelector(getListFavoriteCountries);
  const isLogin = useSelector(getIsAuthentication);

  const handleAddFavoriteCountry = (nameCountry: string[], e: React.SyntheticEvent) => {
    e.stopPropagation();
    dispatch(addFavoriteCountry(nameCountry));
  };

  const handleRemoveFavoriteCountry = (nameCountry: string[], e: React.SyntheticEvent) => {
    e.stopPropagation();
    dispatch(removeFavoriteCountry(nameCountry));
  };

  const handleGoFormLogin = (e: React.SyntheticEvent) => {
    e.stopPropagation();
    dispatch(setModal({ isActiveModal: true, formModal: 'login' }));
  };

  if (isLogin) {
    if (listFavoriteCountries.includes(nameCountry))
      return (
        <BsFillStarFill
          className="star-favorites"
          onClick={(e) => handleRemoveFavoriteCountry([nameCountry], e)}
        />
      );
    else
      return (
        <BsStar className="star" onClick={(e) => handleAddFavoriteCountry([nameCountry], e)} />
      );
  } else {
    return <BsStar className="star" onClick={(e) => handleGoFormLogin(e)} />;
  }
}
