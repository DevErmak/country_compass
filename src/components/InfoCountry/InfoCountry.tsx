// import { AiFillStar } from 'react-icons/ai';
import './info-country.css';
import { ReactComponent as Star } from '../../static/images/star.svg';
import { ReactComponent as StarFavorites } from '../../static/images/star-favorites.svg';

type Props = {
  nameCountry: string;
  nameCapital: string;
  currencies: string;
  region: string;
  languages: string;
  population: string;
  isFavorites: boolean;
};

export default function InfoCountry({
  nameCountry,
  nameCapital,
  currencies,
  region,
  languages,
  population,
  isFavorites,
}: Props) {
  return (
    <div className="container-info-country">
      {isFavorites ? <Star className="star" /> : <StarFavorites className="star-favorites" />}
      <div className="text-info-country">
        <div className="name-country">{nameCountry}</div>
        <div className="name-capital">
          <div className="label-for-info-country">Capital:</div>
          {nameCapital}
        </div>
        <div className="currencies">
          <div className="label-for-info-country">Currencies:</div>
          {currencies}
        </div>
        <div className="region">
          <div className="label-for-info-country">Currencies:</div>

          {region}
        </div>
        <div className="languages">
          <div className="label-for-info-country">Currencies:</div>
          {languages}
        </div>
        <div className="population">
          <div className="label-for-info-country">Population:</div>
          {population}
        </div>
      </div>
    </div>
  );
}
