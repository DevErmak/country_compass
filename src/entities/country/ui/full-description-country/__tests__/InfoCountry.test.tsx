import { renderComponentWithStore } from '../../../shared/lib/test-helpers';
import InfoCountry from '..';

describe('InfoCountry', () => {
  it('renders correctly InfoCountry', () => {
    const { container } = renderComponentWithStore({
      Component: InfoCountry,
      props: {
        fullInfoCountry: {
          nameCountry: '',
          nameCapital: '',
          currencies: '',
          region: '',
          languages: '',
          population: '',
          flags: '',
          flagsAlt: '',
          coatOfArms: '',
        },
      },
      preloadedState: {
        infoCountries: {
          listCountries: [],
          fullInfoCountry: [],
          isFullInfoCountry: false,
          isLoading: false,
          infoErrorResponse: '',
        },
        infoUser: {
          userName: '',
          listFavoriteCountries: [],
          isAuthentication: false,
          infoModal: { isActiveModal: false, formModal: 'login' },
        },
      },
    });
    expect(container).toMatchSnapshot();
  });
});
