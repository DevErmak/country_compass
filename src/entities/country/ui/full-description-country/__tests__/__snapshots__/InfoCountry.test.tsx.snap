// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCountry renders correctly InfoCountry 1`] = `
<div>
  <div
    id="portal"
  />
</div>
`;
